src/dsl.jl

```diff
@@ -352,23 +347,32 @@ function make_reaction_system(ex::Expr; name = :(gensym(:ReactionSystem)))
         combinatoric_ratelaws = true
     end
 
-    # Reads observables.
-    observed_vars, observed_eqs, obs_syms = read_observed_options(
-        options, [species_declared; variables], all_ivs; requiredec)
-
     # Collect species and parameters, including ones inferred from the reactions.
     declared_syms = Set(Iterators.flatten((parameters_declared, species_declared,
-        variables)))
+        variables_declared)))
     species_extracted, parameters_extracted = extract_species_and_parameters!(
         reactions, declared_syms; requiredec)
 
+    # Reads equations (and infers potential variables).
+    # Exlucdes any parameters already extracted (if they also was a variable).
+    designated_syms = [declared_syms; species_extracted]
```

Comment:
Update `designated_syms` with `species_extracted` so these are excluded from the next step.